rosetta-ide/src/main/java/com/regnosys/rosetta/ide/quickfix/RosettaResolveCodeActionService.java

```diff
@@ -0,0 +1,54 @@
+package com.regnosys.rosetta.ide.quickfix;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.xtext.ide.editor.quickfix.DiagnosticResolution;
+import org.eclipse.xtext.ide.editor.quickfix.IQuickFixProvider;
+import org.eclipse.xtext.ide.server.codeActions.ICodeActionService2;
+import org.eclipse.xtext.ide.server.codeActions.ICodeActionService2.Options;
+
+import com.google.common.collect.Iterables;
+import com.regnosys.rosetta.ide.util.CodeActionUtils;
+
+public class RosettaResolveCodeActionService implements IResolveCodeActionService {
+	@Inject
+	CodeActionUtils codeActionUtils;
+	@Inject
+	IQuickFixProvider resolutionProvider;
+	@Inject
+	ICodeActionProvider codeActionProvider;
+
+	@Override
+	public CodeAction getCodeActionResolution(CodeAction codeAction, Options baseOptions) {
+		Diagnostic diagnostic = Iterables.getOnlyElement(codeAction.getDiagnostics());
+
+		ICodeActionService2.Options options = codeActionUtils.createOptionsForSingleDiagnostic(baseOptions, diagnostic);
```

Comment:
This might not work with proposed changes to remove diagnostic for sorting imports